business_objects/record_label_association.py

```diff
@@ -474,6 +474,20 @@ def delete_by_source_id(
     general.flush_or_commit(with_commit)
 
 
+def delete_by_source_id_and_record_ids(
+    project_id: str,
+    information_source_id: str,
+    record_ids: List[str],
+    with_commit: bool = False,
+) -> None:
+    session.query(RecordLabelAssociation).filter(
+        RecordLabelAssociation.project_id == project_id,
+        RecordLabelAssociation.source_id == information_source_id,
+        RecordLabelAssociation.record_id.in_(record_ids),
```

Comment:
in operator with a list of str isn't scaling proof (batching should help)

Reply:
it is only used in batches of 32, so arguably no need for scaling :)